build/gke-test-cluster/cluster.tf

```diff
@@ -0,0 +1,153 @@
+# Copyright 2019 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+provider "google-beta" {
+  zone      = "${lookup(var.cluster, "zone")}"
+}
+
+variable "cluster" {
+  type = "map"
+  default = {
+      "zone" = "us-west1-c"
+      "name" = "test-cluster"
+      "machineType" = "n1-standard-4"
+      "initialNodeCount" = "4"
+      "legacyAbac" = false
+      "project" = "agones"
+  }
+}
+resource "google_container_cluster" "primary" {
+  name     = "${lookup(var.cluster, "name")}"
+  zone     = "${lookup(var.cluster, "zone")}"
+  project = "${lookup(var.cluster, "project")}"
+  provider = "google-beta"
+
+  initial_node_count = "${lookup(var.cluster, "initialNodeCount")}"
+
+  master_auth {
+    username = "admin"
+    password = "supersecretpassword"
```

Comment:
This should also be externalized and perhaps even set to fail if not changed from the default (perhaps we shouldn't even have a default for this).

Reply:
I made a clone of what we have in jinja template:
https://github.com/GoogleCloudPlatform/agones/blob/7368392ed6f2ca4cb42c34f53ef6d6a0730227c3/build/gke-test-cluster/cluster.yml.jinja#L75

Reply:
I added env variable for that, also we can use `password = "${random_string.password.result}"`:
https://medium.com/@tsadoklf/cloud-recipes-use-terraform-to-create-kubernetes-cluster-and-memorystore-redis-instance-on-gcp-d51bb53df46a